scripts/maxmind-db-downloader.js

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env node
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+var path = require('path');
+
+var async = require('async');
+var CronJob = require('cron').CronJob;
+var DEFAULTS = require(path.join('..', 'lib', 'defaults'));
+var fs = require('fs');
+var mkdirp = require('mkdirp');
+var mozlog = require('mozlog');
+var request = require('request');
+var zlib = require('zlib');
+
+// set up mozlog, default is `heka`
+mozlog.config({
+  app: 'fxa-geodb'
+});
+var log = mozlog();
+
+var MaxmindDbDownloader = function () {
+  'use strict';
+
+  this.createTargetDir = function (targetDirName) {
+    targetDirName = targetDirName || DEFAULTS.DEFAULT_TARGET_DIR_NAME;
+    var targetDirPath = path.join(__dirname, '..', targetDirName);
+    // create db folder
+    mkdirp.sync(targetDirPath);
+    // log.info only if not testing
+    // log.error always
+    if (process.env.NODE_ENV !== 'test') {
+      log.info('Download folder is ' + targetDirPath);
+    }
+    return targetDirPath;
+  };
+
+  this.setupDownloadList = function (sourceFilePath, targetDirPath) {
+    sourceFilePath = sourceFilePath || DEFAULTS.DEFAULT_SOURCE_FILE_NAME;
+    targetDirPath = targetDirPath || DEFAULTS.DEFAULT_TARGET_DIR_PATH;
+    // import the list of files to download
+    var sources = require(sourceFilePath);
+    var remainingDownloads = [];
+
+    // push each file-load-function onto the remainingDownloads queue
+    for (var source in sources) {
+      var url = sources[source];
+      if (process.env.NODE_ENV !== 'test') {
+        log.info('Adding ' + url);
+      }
+      // get the file name without the extension
+      var targetFileName = path.parse(source).name;
+      var targetFilePath = path.join(targetDirPath, targetFileName);
+      remainingDownloads.push(this.download(url, targetFilePath));
+      if (process.env.NODE_ENV !== 'test') {
+        log.info('Setting ' + targetFilePath + ' as target file');
+      }
+    }
+    return remainingDownloads;
+  };
+
+  this.download = function (url, targetFilePath) {
+    // closure to separate multiple file-download
+    return function downloadFunctor(callback) {
+      var stream = request(url);
+      stream.pipe(zlib.createGunzip()).pipe(fs.createWriteStream(targetFilePath)).on('finish', function (err) {
+        // forces overwrite, even if file exists already
+        if (err) {
+          log.error(err);
+        } else {
+          // extraction is complete
+          if (process.env.NODE_ENV !== 'test') {
+            log.info('unzip complete');
+          }
+          callback();
+        }
+      });
+    };
+  };
+
+  this.startDownload = function (remainingDownloads) {
+    async.parallel(remainingDownloads, function (err) {
```

Comment:
If [this.download returned a promise](https://github.com/mozilla/fxa-geodb/pull/1/files/75c02a076917d3db24e729d3c31dbf5652208739#r70796126), you could ditch the `async` library and just use [Promise.all](http://bluebirdjs.com/docs/api/promise.all.html). It would also be useful for `startDownload` to return a promise that resolves when all downloads are complete.

`startDownload` seems like a bad name for this function, it seems like it should be `downloadAll` or something similar. And `remainingDownloads` could also be clarified by renaming to something like `downloadPromises`